include/networkit/graph/GraphTools.hpp

```diff
@@ -49,6 +49,29 @@ std::vector<node> invertContinuousNodeIds(std::unordered_map<node,node>& nodeIdM
 Graph restoreGraph(std::vector<node>& invertedIdMap, const Graph& G);
 
 
+/**
+ * Rename nodes in a graph using a callback which translates each old id to a new one.
+ * For each node u in input graph, oldIdToNew(u) < numNodes.
+ *
+ * @param graph Input graph.
+ * @param numNodes Number of nodes in the output graph.
+ * @param oldIdToNew Translate old id to new ones.
+ */
+template <typename UnaryFunc>
+Graph remapNodes(const Graph& graph, count numNodes, UnaryFunc oldIdToNew) {
```

Comment:
I did not review the Curveball and DegreePreservingShuffle code yet, so here is only a first comment: `remapNodes` sounds as if it would modify the Graph in-place. What about renaming this to: `getRemappedGraph()`? This would also be consistent with `getCompactedGraph`. OTOH a name that expresses that this does a lot of work might be appropriate, e.g., `makeRemappedGraph()`.

Reply:
Also: Should this method be able to delete vertices on-the-fly? Maybe with a predicate function argument?

Reply:
Cool idea, I will modify the function it